posydon/popsyn/binarypopulation.py

```diff
@@ -522,18 +523,15 @@ def combine_saved_files(self, absolute_filepath, file_names, **kwargs):
                                  min_itemsize=history_min_itemsize)
                     
                     oneline = pd.read_hdf(f, key='oneline')
-                    if "initially_single_star" in oneline["state_i"].values:
-                        mask= oneline["state_i"]=="initially_single_star"
-                        filtered_data_single = oneline[mask]
-                        filtered_data_binaries = oneline[~mask]
-    
-                        simulated_mass_binaries = np.sum(filtered_data_binaries[["S1_mass_i", "S2_mass_i"]].to_numpy())
-                        simulated_mass_single = np.sum(filtered_data_single[["S1_mass_i"]].to_numpy())
-                        simulated_mass = simulated_mass_binaries + simulated_mass_single
-                    else:
-                        simulated_mass_binaries = np.sum(oneline[["S1_mass_i", "S2_mass_i"]].to_numpy())
-                        simulated_mass_single = 0.0
-                        simulated_mass = simulated_mass_binaries
+                    
+                    # split weight between single and binary stars
+                    mask = oneline["state_i"] == "initially_single_star"
+                    filtered_data_single = oneline[mask]
+                    filtered_data_binaries = oneline[~mask]
+                    
+                    simulated_mass_binaries += np.nansum(filtered_data_binaries[["S1_mass_i", "S2_mass_i"]].to_numpy())
+                    simulated_mass_single += np.nansum(filtered_data_single[["S1_mass_i"]].to_numpy())
+                    simulated_mass += simulated_mass_single + simulated_mass_binaries
```

Comment:
I think, this is wrong now, `simulated_mass_binaries` and `simulated_mass_single` are sums growing with each loop.
`simulated_mass` is supposed to be the sum of the two final sums and not summed again for each loop.
It should read: `simulated_mass = simulated_mass_single + simulated_mass_binaries`, which would be equivalent to `simulated_mass += np.nansum(filtered_data_binaries[["S1_mass_i", "S2_mass_i"]].to_numpy()) + np.nansum(filtered_data_single[["S1_mass_i"]].to_numpy())`